### PR TITLE
attribute.d: remove use of typesafe variadic functions

### PR DESCRIPTION
**Pull request by WalterBright**
Seeing how it goes.

### CODE REVIEW

**Comment by WalterBright** [comment]
Part of https://github.com/dlang/dmd/pull/11124

File: src/core/attribute.d [hunk 1]
```diff
@@ -136,7 +136,7 @@ version (UdaGNUAbiTag) struct gnuAbiTag
 {
     string[] tags;
 
-    this(string[] tags...)
```
Comment (thewilsonator) [inline src/core/attribute.d:139]: Why are these mutually exclusive?

Comment (WalterBright) [reply src/core/attribute.d:139]: Not sure what you mean.

Comment (thewilsonator) [reply src/core/attribute.d:139]: Why does `this(scope string[] tags...)` not work?

Comment (WalterBright) [reply src/core/attribute.d:139]: Still not sure what you mean. `scope` is applied automatically to ... parameters. The replacement for `T[]...` is `scope T[]`.

File: src/core/attribute.d [hunk 2]
```diff
@@ -136,8 +136,14 @@ version (UdaGNUAbiTag) struct gnuAbiTag
...
+    this(scope string[] tags)
     {
         this.tags = tags;
     }
+
+    /* Overloads for backwards compatibility
+     */
+    this(string s1)                       { this.tags = [s1]; }
```
Comment (ibuclaw) [inline src/core/attribute.d:146]: This is a rabbit hole. Maybe it's better moving the variadic part from the parameters to the template definition.

**Comment by RazvanN7** [comment]
@WalterBright It seems that the dmd testsuire is failing:

```
>>> TARGET FAILED: fail_compilation/cpp_abi_tag2.d
FAILED targets:
- fail_compilation/cpp_abi_tag2.d
```

**Comment by WalterBright** [comment]
Replaced by https://github.com/dlang/dmd/pull/14311

**Pull request closed without merging**